eth2/types/src/utils/serde_utils.rs

```diff
@@ -25,13 +29,45 @@ where
     serializer.serialize_str(&hex)
 }
 
+pub fn u32_from_hex_str<'de, D>(deserializer: D) -> Result<u32, D::Error>
+where
+    D: Deserializer<'de>,
+{
+    let s: String = Deserialize::deserialize(deserializer)?;
+    let start = match s.as_str().get(2..) {
```

Comment:
Minor nitpick, we can do this more succinctly with:

```rust
let start = s.as_str().get(2..)
  .ok_or_else(|| D::Error::custom("string length too small"))?;
```

Reply:
Argh of course, the `?`! I sometime forget it when I want an early return... won't forget it next time! :D